src/crypto/rx-slow-hash.c

```diff
@@ -347,3 +347,10 @@ void rx_slow_hash_free_state(void) {
     rx_vm = NULL;
   }
 }
+
+void rx_stop_mining(void) {
+  if (rx_dataset != NULL) {
+    randomx_release_dataset(rx_dataset);
+	rx_dataset = NULL;
+  }
+}
```

Comment:
Bikeshedding, but "mining" seems out of place in this file.
Also, mutex ?

Reply:
It's only called by miner::stop() and only after all mining threads have already been joined, so no mutex is needed.